enhancements/pipeline-tooling.md

```diff
@@ -0,0 +1,267 @@
+
+---
+
+## User Stories
+
+### Migrate to and from each registry format
+
+As an Operator Registry maintainer, I want to continue to push updates to old (appregistry-based) catalogs after having migrated to the new bundle format without having to maintain two repositories.
+
+#### Design Proposal
+
+Add a command to `opm index` and `opm alpha bundle` called `migrate` which:
+
+* takes a collection of bundles migrates these to the old manifest format.
+
+    **Example CLI:**
+    `opm alpha bundle migrate --bundles quay.io/my/bundle:v1,quay.io/my/bundle:v2 --out path/to/my/dir`
+    
+* takes an index and migrate it to the old manifest format.
+    
+    **Example CLI:**
+    `opm index migrate quay.io/my/index --out path/to/my/dir`
+    
+    where `path/to/my/dir` would look like:
+    
+    ```
+    dir
+    ├── bundle
+    │   ├── v1
+    │   │   ├── some.crd.yaml
+    │   │   └── bundle.clusterserviceversion.yaml
+    │   ├── v2
+    │   │   ├── some.crd.yaml
+    │   │   ├── some.other.crd.yaml
+    │   │   ├── bundle.v2.clusterserviceversion.yaml
+    │   └── bundle.package.yaml
+    ```
+
+* takes a manifest directory for a given operator and migrates it to an individual bundle
+
+    Part of this is already implemented in the `opm alpha bundle build` command which takes the manifest for a specific version - for example `path/to/my/dir/bundle/v1` from above:
+    
+    ```
+    dir
+    ├── bundle
+    │   ├── v1
+    │   │   ├── some.crd.yaml
+    │   │   └── bundle.clusterserviceversion.yaml
+    ```
+    
+    and converts that to the new bundle format - including a Dockerfile to push this bundle to a container registry.
+    
+    Q: does it make sense to implement a wrapper for this command that takes a manifest directory instead of a single version to output an index or multiple bundles?
+
+### Download the contents of an index
+
+As an Operator Registry maintainer, Pipeline admin, or QE, I want to download or inspect the contents of a given index image.
+
+#### Design Proposal
+
+Add a command to `opm index` and `opm alpha bundle` called `extract` which extracts:
+
+* the manifest and metadata folders for a given bundle
+
+    **Example CLI:**
+    
+    `opm alpha bundle extract quay.io/my/bundle:v1 --out path/to/my/dir`
+
+* a list of all the bundle images that are in an index
+    
+    **Example CLI 1:**
+    
+    `opm index extract quay.io/my/index --out path/to/my/dir --images-only`
+    
+    **Example CLI 2:** prints to stdout
+    
+    `opm alpha bundle extract quay.io/my/bundle:v1 --image-only`
+
+
+* the manifest and metadata folders for all bundles in a given index
+
+    **Example CLI:**
+    
+    `opm index extract quay.io/my/index --out path/to/my/dir`
+
+
+### Delete latest operator version
+
+As an Operator Registry maintainer, I want to have a way to remove the latest version of an operator (from an index)?
```

Comment:
Do we actually want to implement this if we are moving forward with the implicit semantic versioning proposal? In that case, can't we just make a more generic tool that takes a version and removes that version from the index?